src/makeInfinityAdder.js

```diff
@@ -6,7 +6,23 @@
  */
 
 function makeInfinityAdder() {
-  // write code here
+  let sum = 0;
+
+  const adder = (x) => {
+  const tempSum = sum;
+
+  if (x !== undefined) {
+    sum += x;
+  } else {
+    sum = 0;
+  };
+
+  return x === undefined
+    ? tempSum
+    : adder;
+}
```

Comment:
two similar checks are redundant, i think
its better to `return tempSum;` in else statement and delete ternary

Reply:
if x !== undefined, we increase sum and return function. Else reset sum, and return tempSum

Reply:
I'm a little bit confused) I didn't see this nice solution. Thanks!)